packages/signature-controller/src/SignatureController.ts

```diff
@@ -356,6 +378,12 @@ export class SignatureController extends BaseControllerV2<
     );
   }
 
+  setMessageMetadata(messageId: string, metadata: Json) {
+    this.#setMessageMetadata(this.#messageManager, messageId, metadata);
```

Comment:
Will this not throw if the message is not in the standard messages for example?

Perhaps we could rename `#setMessageMetadata` to `#trySetMessageMetadata` and include a `try catch` and return a boolean so we can skip the subsequent method calls if it returns `true`?

Reply:
Ahh, I see, what you suggest makes total sense 👍🏼